drivers/serial/uart_ns16550_port_x.h

```diff
@@ -130,3 +130,8 @@ static void irq_config_func_@NUM@(const struct device *dev)
 #endif
 
 #endif
+
+/* Include subsequent instances */
+#if @NUM@ < (CONFIG_UART_NS16550_MAX_INSTANCES - 1)
```

Comment:
I'm a bit confused why this technique wouldn't work as a replacement for using CMake configure_file() at all:

https://docs.zephyrproject.org/latest/guides/dts/howtos.html#option-1-create-devices-using-instance-numbers

That would avoid the need for a Kconfig knob.

Reply:
I think one reason is that whatever that expands to can't contain any preprocessor macros of its own (e.g. `#ifdef`), or can it? Take a look at e.g. `irq_config_func_@NUM@()` in `drivers/serial/uart_ns16550_port_x.h`. Could this DT macro be used to generate such functions with all the `#ifdefs` that it contains?

Reply:
Maybe I'm mistaken and maybe the DT macro could be used to generate everything that's in `uart_ns16550_port_x.h`. I just remember that either I tried it or played a thought game with it, and came to the conclusion that it's not feasible.

Reply:
Fair enough, the `#if DT_INST_IRQN(@NUM@) == PCIE_IRQ_DETECT` line in there is tricky to do without its own preprocessor statement.

Reply:
D'oh, I see that still remains even after removing the other Kconfig options; sorry for noise.